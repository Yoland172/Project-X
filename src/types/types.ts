interface FilmItem {
    Title:string;
    Type:string;
    Year:string;
    imdbID:string;
    Poster:string
}

export default FilmItem;