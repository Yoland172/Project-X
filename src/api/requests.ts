import axios from "axios";


export const  getFilms =  async (name:string, page:number)  => {
     console.log(page)
     const result = await axios.get(`${process.env.REACT_APP_OMBD_URL}s=${name}&page=${page}`);
     console.log(result.data);
     return result.data
   
    }

export const getFilmInfo = async(id:string) => {
     const results = await axios.get(`${process.env.REACT_APP_OMBD_URL}i=${id}`);
     console.log(results.data);
     return results.data;
}