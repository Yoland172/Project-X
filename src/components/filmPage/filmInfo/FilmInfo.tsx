import { FilmItemInfoPage } from "../../../types/types";
import styles from './filmInfo.module.scss';
import imdbIcon from '../../../assets/icon/IMDB_Logo_2016.svg.png';
import metacritic from '../../../assets/icon/Metacritic.svg';
import SectionLine from "../../../Helpers/SectionLine/SectionLine";

interface FilmInfoProps {
    filmInfo?: FilmItemInfoPage
}

const FilmInfo = ({ filmInfo }: FilmInfoProps) => {


    return (<>
        <div className={styles.mainContainer}>
            <div className={styles.filmPosterContainer}>
                {filmInfo?.Poster ? <img src={filmInfo.Poster} /> : <></>}
                <div className={styles.ratingList}>
                    <div className={styles.ratingItem}>
                        <h1>IMDB Rating:</h1>
                        <div className={styles.iconWithRating}>
                            <p>{filmInfo?.imdbRating} / <span>10</span>
                                {/* <span className={styles.votes}>
                        {filmInfo!.imdbVotes.length > 4 ? `${filmInfo?.imdbVotes.substring(0,4)} k` : filmInfo?.imdbVotes } // add
                    </span> */}
                            </p>
                            <img src={imdbIcon} />
                        </div>
                    </div>
                    {/* <div className={styles.ratingItem}>
                        <h1>TOMATOMETR:</h1>
                        <div className={styles.iconWithRating}>
                            <p>73%</p>  <img src={tomatoIcon} />  add TOMATOMETR.........................................
                        </div>

                    </div> */}
                    {filmInfo?.Metascore != "N/A" ? <div className={styles.ratingItem}>
                        <h1>METACRITIC:</h1>
                        <div className={styles.iconWithRating}>
                            <p>{filmInfo?.Metascore} / <span>100</span></p>  <img src={metacritic} />
                        </div>

                    </div> : <></>
                    }
                </div>

            </div>
            <div className={styles.filmInfoContainer}>
                <h1 className={styles.title}>
                    {filmInfo?.Title}
                </h1>
                <div className={styles.addInfo}>
                    <div className={styles.runtime}>
                        <p>Runtime:<span>{filmInfo?.Runtime}</span></p>
                    </div>
                    <div className={styles.genre}>
                        <p>Genre:<span>{filmInfo?.Genre}</span></p>
                    </div>
                    <div className={styles.year}>
                        <p>Year:<span>{filmInfo?.Year}</span></p>
                    </div>
                    <div className={styles.type}>
                        <p>Type:<span>{filmInfo?.Type}</span></p>
                    </div>
                    <div className={styles.plot}>
                        <p>
                            {filmInfo?.Plot}
                        </p>
                    </div>
                </div>
                <SectionLine />


                {filmInfo?.Awards != "N/A" ?
                    <div className={styles.oscarAwardsContainer}>
                        <div className={styles.goldContainer}>
                        </div>
                        <div>
                        </div>
                        <div className={styles.awardsText}>
                            <p>{filmInfo?.Awards}</p>
                        </div>
                    </div>
                    : <></>
                }
                <div className={styles.addInfoContainer}>
                    <div className={styles.directorsTest}>
                        <div className={styles.verticalLine}></div>
                        <div className={styles.textContainer}>
                            <h1>{filmInfo?.Director}</h1>
                            <p>Director</p>
                        </div>
                    </div>
                    <div className={styles.directorsTest}>
                        <div className={styles.verticalLine}></div>
                        <div className={styles.textContainer}>
                            <h1>{filmInfo?.Rated}</h1>
                            <p>Rated</p>
                        </div>
                    </div>
                    <div className={styles.directorsTest}>
                        <div className={styles.verticalLine}></div>
                        <div className={styles.textContainer}>
                            <h1>{filmInfo?.Language}</h1>
                            <p>Language</p>
                        </div>
                    </div>
                    <div className={styles.directorsTest}>
                        <div className={styles.verticalLine}></div>
                        <div className={styles.textContainer}>
                            <h1>{filmInfo?.Released}</h1>
                            <p>Released</p>
                        </div>

                    </div>
                    <div className={styles.directorsTest}>
                        <div className={styles.verticalLine}></div>
                        <div className={styles.textContainer}>
                            <h1>{filmInfo?.Country}</h1>
                            <p>Country</p>
                        </div>
                    </div>
                    {filmInfo?.BoxOffice != "N/A" ?
                        <div className={styles.directorsTest}>
                            <div className={styles.verticalLine}></div>   {/*make check for all blocks */}
                            <div className={styles.textContainer}>
                                <h1>{filmInfo?.BoxOffice}</h1>
                                <p>Box Office</p>
                            </div>
                        </div>
                        : <></>

                    }
                </div>
                {/*filmInfo?.Writer != "N/A" ?
                <div className={styles.directorsTest}>
                        <h1>{filmInfo?.Writer}</h1>
                        <p>Writer</p>
                    </div>
                    : <></> */
                }

            </div>

                
        </div>
        </>
    )
}

export default FilmInfo;

