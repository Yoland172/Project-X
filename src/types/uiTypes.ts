export interface FilmItemMainPage {
  Title: string;
  Type: string;
  Year: string;
  imdbID: string;
  Poster: string;
}
export interface IconProps {
  width: number;
  height: number;
}


export interface IconThemeProps {
  width: number;
  height: number;
  className?: string;
}
