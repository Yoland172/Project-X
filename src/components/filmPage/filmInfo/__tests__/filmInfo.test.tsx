import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import FilmInfo from "../FilmInfo";
import { FilmItemInfo } from "../../../../api/types";

describe("FilmInfo component", () => {
  const filmInfo: FilmItemInfo = {
    Title: "The Batman",
    Year: "2022",
    Rated: "PG-13",
    Released: "04 Mar 2022",
    Runtime: "176 min",
    Genre: "Action, Crime, Drama",
    Director: "Matt Reeves",
    Writer: "Matt Reeves, Peter Craig, Bob Kane",
    Actors: "Robert Pattinson, ZoÃ« Kravitz, Jeffrey Wright",
    Plot: "When a sadistic serial killer begins murdering key political figures in Gotham, Batman is forced to investigate the city's hidden corruption and question his family's involvement.",
    Language: "English, Spanish, Latin, Italian",
    Country: "United States",
    Awards: "Nominated for 3 Oscars. 35 wins & 165 nominations total",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BM2MyNTAwZGEtNTAxNC00ODVjLTgzZjUtYmU0YjAzNmQyZDEwXkEyXkFqcGdeQXVyNDc2NTg3NzA@._V1_SX300.jpg",
    Ratings: [
      {
        Source: "Internet Movie Database",
        Value: "7.8/10",
      },
      {
        Source: "Rotten Tomatoes",
        Value: "85%",
      },
      {
        Source: "Metacritic",
        Value: "72/100",
      },
    ],
    Metascore: "72",
    imdbRating: "7.8",
    imdbVotes: "753,503",
    imdbID: "tt1877830",
    Type: "movie",
    DVD: "19 Apr 2022",
    BoxOffice: "$369,345,583",
    Production: "N/A",
    Website: "N/A",
    Response: "True",
  };

  test("Component matches shapshot", () => {
    //Act
    const tree = renderer.create(<FilmInfo filmInfo={filmInfo} />).toJSON();

    //Assert
    expect(tree).toMatchSnapshot();
  });

  test("elements of page render", () => {
    render(<FilmInfo filmInfo={filmInfo} />);
    const img = screen.queryByAltText("FilmPoster");
    const metacritic = screen.queryByAltText("Metacritic");
    const runtime = screen.getByText(/Runtime/i);
    const genre = screen.getByText(/Genre/i);
    const year = screen.getByText(/Year/i);
    const director = screen.getByText(/Director/i);
    const rated = screen.getByText(/Rated/i);
    const language = screen.getByText(/Language/i);
    const released = screen.getByText(/Released/i);
    const country = screen.getByText(/Country/i);
    const boxOffice = screen.getByText(/Box/i);

    //Assert
    expect(img).toBeInTheDocument();
    expect(metacritic).toBeInTheDocument();
    expect(runtime).toBeInTheDocument();
    expect(genre).toBeInTheDocument();
    expect(year).toBeInTheDocument();
    expect(director).toBeInTheDocument();
    expect(rated).toBeInTheDocument();
    expect(language).toBeInTheDocument();
    expect(released).toBeInTheDocument();
    expect(country).toBeInTheDocument();
    expect(boxOffice).toBeInTheDocument();
  });

  test("elements of page do not render", () => {
    //Act
    render(<FilmInfo />);
    const img = screen.queryByAltText("FilmPoster");
    const metacritic = screen.queryByAltText("queryByAltText");
    const runtime = screen.queryByText(/Runtime/i);
    const genre = screen.queryByText(/Genre/i);
    const year = screen.queryByText(/Year/i);
    const director = screen.queryByText(/Director/i);
    const rated = screen.queryByText(/Rated/i);
    const language = screen.queryByText(/Language/i);
    const released = screen.queryByText(/Released/i);
    const country = screen.queryByText(/Country/i);
    const boxOffice = screen.queryByText(/Box Office/i);

    //Assert
    expect(img).toBeNull();
    expect(metacritic).toBeNull();
    expect(runtime).toBeNull();
    expect(genre).toBeNull();
    expect(year).toBeNull();
    expect(director).toBeNull();
    expect(rated).toBeNull();
    expect(language).toBeNull();
    expect(released).toBeNull();
    expect(country).toBeNull();
    expect(boxOffice).toBeNull();
  });
});
