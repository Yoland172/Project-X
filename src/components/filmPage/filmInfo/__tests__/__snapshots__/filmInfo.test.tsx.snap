// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfo component Component matches shapshot 1`] = `
<div
  className="mainContainer"
>
  <div
    className="filmPosterContainer"
  >
    <img
      alt="FilmPoster"
      src="https://m.media-amazon.com/images/M/MV5BM2MyNTAwZGEtNTAxNC00ODVjLTgzZjUtYmU0YjAzNmQyZDEwXkEyXkFqcGdeQXVyNDc2NTg3NzA@._V1_SX300.jpg"
    />
    <div
      className="ratingList"
    >
      <div
        className="ratingItem"
      >
        <h1>
          IMDB Rating:
        </h1>
        <div
          className="iconWithRating"
        >
          <p>
            7.8
             / 
            <span>
              10
            </span>
          </p>
          <img
            alt="IMDB"
            src="IMDB_Logo_2016.svg.png"
          />
        </div>
      </div>
      <div
        className="ratingItem"
      >
        <h1>
          METACRITIC:
        </h1>
        <div
          className="iconWithRating"
        >
          <p>
            72
             / 
            <span>
              100
            </span>
          </p>
          <img
            alt="Metacritic"
            src="Metacritic.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="filmInfoContainer"
  >
    <h1
      className="title"
    >
      The Batman
    </h1>
    <div
      className="addInfo"
    >
      <div
        className="runtime"
      >
        <p>
          Runtime:
          <span>
            176 min
          </span>
        </p>
      </div>
      <div
        className="genre"
      >
        <p>
          Genre:
          <span>
            Action, Crime, Drama
          </span>
        </p>
      </div>
      <div
        className="year"
      >
        <p>
          Year:
          <span>
            2022
          </span>
        </p>
      </div>
      <div
        className="type"
      >
        <p>
          Type:
          <span>
            movie
          </span>
        </p>
      </div>
      <div
        className="plot"
      >
        <p>
          When a sadistic serial killer begins murdering key political figures in Gotham, Batman is forced to investigate the city's hidden corruption and question his family's involvement.
        </p>
      </div>
    </div>
    <div
      className="line"
    />
    <div
      className="oscarAwardsContainer"
    >
      <div
        className="oscarAwards"
      />
      <div />
      <div
        className="awardsText"
      >
        <p>
          Nominated for 3 Oscars. 35 wins & 165 nominations total
        </p>
      </div>
    </div>
    <div
      className="addInfoContainer"
    >
      <div
        className="directorsTest"
      >
        <div
          className="verticalLine"
        />
        <div
          className="textContainer"
        >
          <h1>
            Matt Reeves
          </h1>
          <p>
            Director
          </p>
        </div>
      </div>
      <div
        className="directorsTest"
      >
        <div
          className="verticalLine"
        />
        <div
          className="textContainer"
        >
          <h1>
            PG-13
          </h1>
          <p>
            Rated
          </p>
        </div>
      </div>
      <div
        className="directorsTest"
      >
        <div
          className="verticalLine"
        />
        <div
          className="textContainer"
        >
          <h1>
            English, Spanish, Latin, Italian
          </h1>
          <p>
            Language
          </p>
        </div>
      </div>
      <div
        className="directorsTest"
      >
        <div
          className="verticalLine"
        />
        <div
          className="textContainer"
        >
          <h1>
            04 Mar 2022
          </h1>
          <p>
            Released
          </p>
        </div>
      </div>
      <div
        className="directorsTest"
      >
        <div
          className="verticalLine"
        />
        <div
          className="textContainer"
        >
          <h1>
            United States
          </h1>
          <p>
            Country
          </p>
        </div>
      </div>
      <div
        className="directorsTest"
      >
        <div
          className="verticalLine"
        />
         
        <div
          className="textContainer"
        >
          <h1>
            $369,345,583
          </h1>
          <p>
            Box Office
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
